src/utils/query.js

```diff
@@ -166,9 +169,13 @@ export function firestoreRef(firebase, meta) {
       'Reference cannot contain both Collection and CollectionGroup.',
     );
   }
+  const { globalDataConvertor } =
+    (firebase && firebase._ && firebase._.config) || {};
 
+  if (path) ref = ref.collection(path);
   if (collection) ref = ref.collection(collection);
   if (collectionGroup) ref = ref.collectionGroup(collectionGroup);
+  if (id) ref = ref.doc(id);
```

Comment:
What is the difference between this and `doc`?

Reply:
I really favor homogeneous APIs. Reduce the complexity but reducing the terminology. The mutate is a great example:
```
// single
{collection: 'coll', doc: 'doc', data: {a:1}}
// batch
[{collection: 'coll', doc: 'doc', data: {a:1}}]
// write transaction: 
{write: {collection: 'coll', doc: 'doc', data: {a:1}}}
// read/write transaction:
{read: {myRead: {collection:'coll', doc:'doc'}} 
  write: ({myRead}) => ({collection: 'myCollection', doc: 'myDoc', data: {a:1}})}
```
You notice it's just the same collection/doc/data pattern. Once you see that it's trivial to switch between a type of save.

But when you look at the document references you have a different format. Firestore refs have `id`, and `path` to describe the same thing. So the question is does that extra terminology help communicate a boundary or is it just waste?

Why can't I just take a JSON object that comes out of Firestore and send it back (or a partial of it)?

In the Query Snapshot and DocumentReferenceSnapshots coming out of the listener add `id` and `path` to all data. First it's backward compatible with the data/ordered reducers. Second it's using DocumentReference terms of `id` and `path` to align closely to the official Firestore docs as much as possible. But here's the big question. How do we benefit from query and saving having different terms/API? I don't think we do.

So I support collection/doc as much as possible to make it easy to use the new reducer. The only change devs need to understand is the FieldValue (since I can't reflect then reliably). But by supporting the `id` and `path` for everything it makes the API so much simpler.

```js
const tasks = useSelector(taskSelector);
const {id, path, name} = tasks[0];

// with the current API
dispatch(sendMutate({doc: id, collection: path, data: { name } }));

// with the newer homogeneous API
dispatch(sendMutate({id, path, name}));

// Also mutate function also supports both the current API or the new homogeneous API
// {read: {myRead: {path, id}} 
//  write: ({myRead}) => ({path, id, ...rest})}
```

So in short, `id` and `doc` are interchangeable. `path` and `collection` are mostly interchangeable, but collection supports an object notation where path is the same URL-style. By moving everything to support `id` and `path` the public API is so small, clear and simple.